File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
import sys
input = sys.stdin.readline

N = int(input())
count = -1

if N % 5 == 0:
    count = N / 5
else:
    for i in range(0, N // 3):
        if (N - (3 * i)) % 5 == 0:
            fiveCount = (N - (3 * i)) // 5
            count = i + fiveCount
            break
        elif N % 3 == 0:
            count = N // 3

print(int(count))

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int count = 0;


        if (n == 3)
            count = 1;

        if (n == 4)
            count = -1;

        if (n % 5 == 0) {
            count = n / 5;
        } else {

            for (int i = 1; i < (n / 3); i++)
                if ((n - (3 * i)) % 5 == 0) {

                    int fiveCount = (n - 3 * i) / 5;

                    count = i + fiveCount;

                    break;

                } else if (n % 3 == 0) {
                    count = n / 3;

                } else
                    count = -1;
        }
        System.out.println(count);
    }
}